relay-server/src/services/buffer/envelope_buffer/mod.rs

```diff
@@ -397,62 +228,55 @@ where
             });
     }
 
-    /// Returns `true` if the underlying storage has the capacity to store more envelopes.
+    /// Returns `true` if the underlying storage has the capacity to store more envelopes, false
+    /// otherwise.
     pub fn has_capacity(&self) -> bool {
-        self.stack_provider.has_store_capacity()
+        match &self.envelope_repository {
+            EnvelopeRepository::Memory(repository) => repository.has_store_capacity(),
+            EnvelopeRepository::SQLite(repository) => repository.has_store_capacity(),
+        }
```

Comment:
I would move this `match` to `EnvelopeRepository`, as we did with the other methods.